### PR TITLE
Upgrade Angora to most up-to-date. Changed some build script so Angora compiles

### PR DESCRIPTION
**Pull request by DataCorrupted**
Angora has migrated to LLVM11 and some bugs are fixed (See [PR](https://github.com/AngoraFuzzer/Angora/pull/113)).
Migrate Magma's Angora to the most updated version.

Signed-off-by: Peter Rong <PeterRong96@gmail.com>

### CODE REVIEW

**Comment by adrianherrera** [comment]
Nice, thanks! I'll give it a spin today and see how it goes

File: fuzzers/angora/instrument.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ export CXX="$FUZZER/repo/bin/angora-clang++"
     export LDFLAGS="$LDFLAGS -L$OUT"
 
     export USE_FAST=1
-    export LIBS="$LIBS -l:angora_driver.o"
+    export LIBS="$LIBS -l:angora_driver.o -lstdc++ -lc++"
```
Comment (adrianherrera) [inline fuzzers/angora/instrument.sh:25]: Is there a reason for adding `-lc++` here? Mine builds fine without it

File: fuzzers/angora/instrument.sh [hunk 2]
```diff
@@ -46,13 +46,14 @@ export CXX="$FUZZER/repo/bin/angora-clang++"
...
     done
+    cat $FUZZER/src/extra_abilist.txt >> "$TARGET/repo/abilist.txt"
 )
 
 # Build track target
 (
     export OUT="$OUT/angora-track"
     export LDFLAGS="$LDFLAGS -L$OUT -L$FUZZER/repo/bin/lib"
-    export LIBS="$LIBS -l:angora_driver.o"
+    export LIBS="$LIBS -l:angora_driver.o -lstdc++ -lc++"
```
Comment (adrianherrera) [inline fuzzers/angora/instrument.sh:55]: Same here re. `-lc++`

File: fuzzers/angora/src/extra_abilist.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+fun:make_fcontext=uninstrumented
```
Comment (adrianherrera) [inline fuzzers/angora/src/extra_abilist.txt]: Where does this file come from? Is this an Angora thing?

File: fuzzers/angora/instrument.sh (hunk 1 above)
Comment (DataCorrupted) [reply fuzzers/angora/instrument.sh:25]: Only `-lstdc++` is required. Angora had some linking problems when compiling OpenSSL. I must have forgotten to remove `-lc++`.

File: fuzzers/angora/src/extra_abilist.txt (hunk 3 above)
Comment (DataCorrupted) [reply fuzzers/angora/src/extra_abilist.txt]: PHP includes Zend now, which has some assembly code in `repo/Zend/asm/`. These codes introduce functions like `make_fcontext`, `jump_fcontext`. They originate in the boost library but PHP seems to have assembly versions of them.

These functions are not instrumented nor modeled by Angora, thus ignored.

Comment (adrianherrera) [reply fuzzers/angora/src/extra_abilist.txt]: Ah ok cool, makes sense! Is it possible to do something to what we already to do handle inline asm in openssl? See https://github.com/HexHive/magma/blob/v1.2/targets/openssl/src/abilist.txt

If this is specific to php, I suggest moving into the target.

File: fuzzers/angora/instrument.sh (hunk 1 above)
Comment (adrianherrera) [reply fuzzers/angora/instrument.sh:25]: Ok cool. Please remove and update your PR

File: fuzzers/angora/src/extra_abilist.txt (hunk 3 above)
Comment (DataCorrupted) [reply fuzzers/angora/src/extra_abilist.txt]: Sure, let me see if I can get it done tonight. (Somehow Angora can take ~1 hour to compile PHP)

**Comment by adrianherrera** [comment]
Can you please squash the commits and I can merge!

**Comment by DataCorrupted** [comment]
> Can you please squash the commits and I can merge!

Done

**Comment by adrianherrera** [comment]
Thanks for doing that!

Sorry, I just realized this PR was made against the v1.2 branch. Can you please make it against the dev branch and then I'll merge it. Sorry for the hassle!!! (I should add a CONTRIBUTING.md file that describes this)

**Comment by DataCorrupted** [comment]
> c

Done

**Comment by DataCorrupted** [comment]
> Thanks for doing that!
> 
> Sorry, I just realized this PR was made against the v1.2 branch. Can you please make it against the dev branch and then I'll merge it. Sorry for the hassle!!! (I should add a CONTRIBUTING.md file that describes this)

Done

**Comment by DataCorrupted** [comment]
Hi @adrianherrera I don't think this is merged before you close it. Do you still want it?

(Sorry I forgot about this PR too)

**Pull request closed without merging**